plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/CorruptedIcebergTableHandle.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.trino.spi.TrinoException;
+import io.trino.spi.connector.ConnectorTableHandle;
+import io.trino.spi.connector.SchemaTableName;
+
+import static java.util.Objects.requireNonNull;
+
+public record CorruptedIcebergTableHandle(SchemaTableName schemaTableName, TrinoException originalException)
+        implements ConnectorTableHandle
+{
+    public CorruptedIcebergTableHandle
+    {
+        requireNonNull(schemaTableName, "schemaTableName is null");
+        requireNonNull(originalException, "originalException is null");
```

Comment:
The name `originalException` in the context of a table handle is rather misleading.

Just another option to take into consideration: `originalException` -> `corruptionCause`

Reply:
I was following delta-lake for the name https://github.com/trinodb/trino/blob/c23ec5e7d11edb471e8b26dd68dac0a067322b50/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/CorruptedDeltaLakeTableHandle.java#L30